.github/workflows/main.yml

```
@@ -95,6 +95,22 @@ jobs:
         env:
           CI_COMMIT_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
 
+  test-ssr:
+    needs: setup
+    runs-on: ubuntu-latest
+
+    steps:
```

You'll have to restore the setup as well.
```suggestion
    steps:
      - name: Restore setup
        uses: actions/cache@v4
        with:
          path: ./*
          key: ${{ runner.os }}-${{ github.ref }}-${{ github.sha }}-setup

```